packages/list-plugins/lib/limiting/singleton.js

```diff
@@ -0,0 +1,43 @@
+const { composeHook, composeAccess } = require('../utils');
+
+exports.singleton = ({ preventDelete = true }) => ({
+  key,
+  keystone,
+  access,
+  hooks = {},
+  adminConfig = {},
+  ...rest
+}) => {
+  const newResolveInput = async ({ resolvedData, operation }) => {
+    if (operation === 'create') {
+      const list = keystone.getListByKey(key);
```

Comment:
this and `executeQuery` scenario is enabled by passing `key` and `keystone` to the plugin config. These are discarded by the List constructor anyways.

Reply:
just of this plugin, the keystone parameter can be reverted and We could construct the query using `all${key}Meta` instead of using `listQueryMetaName` and use `action: { queery }` of hook argument rather than `keystone.executeQuery`